### PR TITLE
Ensure the cleanup in testAutoFollowPatterns is always happening

### LINKED ISSUE #85147: [CI] FollowIndexSecurityIT testFollowIndex failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/iuuchmzstcy7m/tests/:x-pack:plugin:ccr:qa:security:follow-cluster/org.elasticsearch.xpack.ccr.FollowIndexSecurityIT/testFollowIndex

**Reproduction line:**
`./gradlew ':x-pack:plugin:ccr:qa:security:follow-cluster' --tests "org.elasticsearch.xpack.ccr.FollowIndexSecurityIT.testFollowIndex" -Dtests.seed=7E4A8B8D7A291AAB -Dtests.locale=zh-Hant-TW -Dtests.timezone=Asia/Seoul -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.ccr.FollowIndexSecurityIT&tests.test=testFollowIndex

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: <1>
     but: was <2>

  at __randomizedtesting.SeedInfo.seed([7E4A8B8D7A291AAB:9D2644BF43912321]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.ccr.FollowIndexSecurityIT.testFollowIndex(FollowIndexSecurityIT.java:69)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by idegtiarenko**
In case of failure testAutoFollowPatterns will leave autofollowing
pattern running that would case assertion failure in other test cases

Closes: #85147

### CODE REVIEW

File: x-pack/plugin/ccr/qa/security/src/test/java/org/elasticsearch/xpack/ccr/FollowIndexSecurityIT.java [hunk 1]
```diff
@@ -175,20 +172,21 @@ public void testAutoFollowPatterns() throws Exception {
...
+            assertThat(indexExists(disallowedIndex), is(false));
+            assertBusy(() -> {
+                verifyCcrMonitoring(allowedIndex, allowedIndex);
+                verifyAutoFollowMonitoring();
+            }, 30, TimeUnit.SECONDS);
+        } finally {
+            // Cleanup by deleting auto follow pattern and pause following:
+            assertOK(client().performRequest(new Request("DELETE", "/_ccr/auto_follow/test_pattern")));
+            pauseFollow(client(), allowedIndex);
+        }
```
Comment (idegtiarenko) [inline x-pack/plugin/ccr/qa/security/src/test/java/org/elasticsearch/xpack/ccr/FollowIndexSecurityIT.java:189]: Actual fix

File: x-pack/plugin/ccr/qa/security/src/test/java/org/elasticsearch/xpack/ccr/FollowIndexSecurityIT.java [hunk 2]
```diff
@@ -175,20 +172,21 @@ public void testAutoFollowPatterns() throws Exception {
...
+                verifyDocuments(allowedIndex, 5, "*:*");
+            }, 30, TimeUnit.SECONDS);
+            assertThat(indexExists(disallowedIndex), is(false));
+            assertBusy(() -> {
+                verifyCcrMonitoring(allowedIndex, allowedIndex);
+                verifyAutoFollowMonitoring();
+            }, 30, TimeUnit.SECONDS);
+        } finally {
+            // Cleanup by deleting auto follow pattern and pause following:
+            assertOK(client().performRequest(new Request("DELETE", "/_ccr/auto_follow/test_pattern")));
```
Comment (tlrx) [inline x-pack/plugin/ccr/qa/security/src/test/java/org/elasticsearch/xpack/ccr/FollowIndexSecurityIT.java:187]: Can we use `deleteAutoFollowPattern`?

File: x-pack/plugin/ccr/qa/security/src/test/java/org/elasticsearch/xpack/ccr/FollowIndexSecurityIT.java [hunk 3]
```diff
@@ -175,20 +172,21 @@ public void testAutoFollowPatterns() throws Exception {
...
-        pauseFollow(client(), allowedIndex);
+        try {
+            assertBusy(() -> {
+                ensureYellow(allowedIndex);
+                verifyDocuments(allowedIndex, 5, "*:*");
+            }, 30, TimeUnit.SECONDS);
+            assertThat(indexExists(disallowedIndex), is(false));
+            assertBusy(() -> {
+                verifyCcrMonitoring(allowedIndex, allowedIndex);
+                verifyAutoFollowMonitoring();
```
Comment (tlrx) [inline x-pack/plugin/ccr/qa/security/src/test/java/org/elasticsearch/xpack/ccr/FollowIndexSecurityIT.java:183]: Can we put each verification method into its own assertBusy()?

**Review by tlrx — APPROVED** [review]
LGTM

File: x-pack/plugin/ccr/qa/security/src/test/java/org/elasticsearch/xpack/ccr/FollowIndexSecurityIT.java (hunk 2 above)
Comment (idegtiarenko) [reply x-pack/plugin/ccr/qa/security/src/test/java/org/elasticsearch/xpack/ccr/FollowIndexSecurityIT.java:187]: Good point. This and other requests could be simplified

**Comment by idegtiarenko** [comment]
Thank you, @tlrx!

**Comment by idegtiarenko** [comment]
@elasticmachine please run elasticsearch-ci/part-1

**Pull request merged**